src/ralph/backends/data/base.py

```diff
@@ -223,3 +223,149 @@ def write(  # pylint: disable=too-many-arguments
                 `ignore_errors` is set to `False`.
             BackendParameterException: If a backend argument value is not valid.
         """
+
+
+class BaseAsyncDataBackend(ABC):
+    """Base data backend interface."""
+
+    name = "base"
+    query_model = BaseQuery
+    default_operation_type = BaseOperationType.INDEX
+    settings_class = BaseDataBackendSettings
+
+    @abstractmethod
+    def __init__(self, settings: settings_class = None):
+        """Instantiates the data backend.
```

Comment:
```suggestion
        """Instantiate the async data backend.
```
Same suggestion about imperative docstring-style for other methods